### PR TITLE
[Merged by Bors] - fix(library/init/meta/interactive): fix let with coe_sort

### PR DESCRIPTION
**Pull request by gebner**

### CODE REVIEW

File: tests/lean/run/let_coe_sort.lean [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+begin
+  let x : foo.bar := (),
+  have y : foo.bar := (),
+  trivial
+end
+
+example : true :=
+let x : foo.bar := () in
+have y : foo.bar, from (),
+trivial
```
Comment (eric-wieser) [inline tests/lean/run/let_coe_sort.lean:18]: github seems to suggest
```suggestion
trivial

```

Comment (gebner) [reply tests/lean/run/let_coe_sort.lean]: If it makes Lord Octocat happy....

File: src/frontends/lean/elaborator.cpp [hunk 2]
```diff
@@ -3405,6 +3405,7 @@ expr elaborator::visit_pi(expr const & e) {
 expr elaborator::visit_let(expr const & e, optional<expr> const & expected_type) {
     expr ref = e;
     expr new_type  = visit(let_type(e), none_expr());
+    new_type = ensure_type(new_type, let_type(e));
```
Comment (eric-wieser) [inline src/frontends/lean/elaborator.cpp:3408]: Every other call to `ensure_type` in this file seems to call `get_ref_for_child` in order to obtain the second argument to this call:
https://github.com/leanprover-community/lean/blob/508db1acd2249ac7bc0654867ab73a6d5c0a6408/src/frontends/lean/elaborator.cpp#L3387-L3389

Is there a reason that doesn't make sense here?

File: tests/lean/run/let_coe_sort.lean [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+instance : has_coe_to_sort foo :=
+⟨Type, λ _, unit⟩
+
+set_option pp.all true
+
+example : true :=
+begin
+  let x : foo.bar := (),
+  have y : foo.bar := (),
```
Comment (eric-wieser) [inline tests/lean/run/let_coe_sort.lean:11]: Is it worth adding a test for `suffices` too?

File: src/frontends/lean/elaborator.cpp (hunk 2 above)
Comment (gebner) [reply src/frontends/lean/elaborator.cpp]: I guess it's a reasonable precaution.

File: tests/lean/run/let_coe_sort.lean (hunk 3 above)
Comment (gebner) [reply tests/lean/run/let_coe_sort.lean:11]: Feel free to add one.

File: tests/lean/run/let_coe_sort.lean [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  trivial
```
Comment (eric-wieser) [inline tests/lean/run/let_coe_sort.lean:12]:
```suggestion
  suffices y : foo.bar,
  trivial,
  exact (),
```

**Pull request closed without merging**